testing/test_terminal.py

```diff
@@ -1633,9 +1633,9 @@ def test_failure():
     result.stdout.no_fnmatch_line("*= short test summary info =*")
 
 
-@pytest.fixture(scope="session")
-def tr() -> TerminalReporter:
-    config = _pytest.config._prepareconfig()
+@pytest.fixture(scope="function")
+def tr(pytester: Pytester) -> TerminalReporter:
+    config = pytester.parseconfig()
     return TerminalReporter(config)
```

Comment:
Apparently this change helped with aforementioned issue (see description) - builds are passing even with migrated `test_config.py`.
And I don't see "**testing/test_terminal.py:1642: PytestWarning: ***** 2 FD leakage detected**" in the [logs](https://github.com/pytest-dev/pytest/runs/1514494743?check_suite_focus=true) anymore.

@blueyed - could you take a look at it? Perhaps you will be able to grasp what's going on here (I'm pretty new to this).